File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Clothing.java
package com.logistics.Components.Product;

public class Clothing extends Apparel {

    private String style;

    public Clothing() {
        super();
        style = "";
    }

    public Clothing(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String material, String style) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, material);
        this.style = style;
    }

    // Copy Constructor
    public Clothing(Clothing other) {
        super(other);
        this.style = other.style;
    }

    // Clone method
    @Override
    public Clothing clone() {
        return new Clothing(this);
    }

    // Getters
    public String getStyle() { return style; }

    // Setters
    public void setStyle(String style) { this.style = style; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + "\nStyle: " + style;
    }
}

File: Task 1/C++/Components/Product/Clothing.cpp
#include "Clothing.h"

Clothing::Clothing() : Apparel(){
    style = "";
}

Clothing::Clothing(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string material, std::string style) 
            : Apparel(name, cost, supplier, discount_amount, packing_units, cost_per_unit, material){
    this->style = style;
}

Clothing::Clothing(const Clothing& other) : Apparel(other) {
    this->style = other.style;
}

std::string Clothing::display_product_info(){
    return Apparel::display_product_info() +
            "\nStyle: " +style;
}